transaction-decoder-tool/src/test/kotlin/linea/test/BlocksFetcher.kt

```diff
@@ -0,0 +1,77 @@
+package linea.test
+
+import build.linea.web3j.domain.toWeb3j
+import io.vertx.core.Vertx
+import linea.domain.Block
+import linea.web3j.toDomain
+import net.consensys.linea.BlockParameter.Companion.toBlockParameter
+import net.consensys.linea.async.AsyncRetryer
+import net.consensys.linea.async.toSafeFuture
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.Logger
+import org.web3j.protocol.Web3j
+import tech.pegasys.teku.infrastructure.async.SafeFuture
+import java.util.concurrent.atomic.AtomicLong
+import kotlin.time.Duration.Companion.milliseconds
+
+class BlocksFetcher(
```

Comment:
IIUC purpose of this class is to fetch blocks in chunks and execute some user logic. So it's kind of `BlockHandlingRunner` or something, not only fetcher

Reply:
I would rather keep the name, 90%+ of the code is logic to actually fetch the blocks, then the `consumer` is that actually runs that logic, which can be sync or deferred like in the validator for compression and decompression.